### ISSUE TITLE
False positive lines_longer_than_80_chars on lines with CRLF endings

### ISSUE THREAD

**Issue by lexaknyazev** [dart-lang/linter#1071]
A line of code containing exactly 80 chars passes linter with LF ending but raises `lines_longer_than_80_chars` with CRLF ending.

Dart SDK 2.0.0-dev.68.0

**Comment by a14n** [comment]
@bwilkerson, @pq : what's the best way to get the char at a given offset when you have a `CompilationUnit` ?

**Comment by bwilkerson** [comment]
There isn't any good way at the moment, and the way that works today will likely be broken in the near future. That said, the following ought to work:
```
int charAt(CompilationUnit unit, int offset) {
  CompilationUnitElement element = unit.element;
  String contents = element.context.getContents(element.source).data;
  return contents.codeUnitAt(offset);
}
```
